### PR TITLE
fix(backend/executor): bind the scheduler's RPC port before the embedding backfill

### PR DESCRIPTION
**Pull request by Pwuts**
### Why / What / How

The scheduler now accepts RPC connections as soon as it starts, instead of ~4.5 minutes later.

The fix is [@ntindle](https://github.com/ntindle)'s, from #14350 ("fix(backend): preserve startup indexing without blocking readiness"). That PR is layer 11 of a twelve-deep `codex/ci-*` stack and cannot land until the whole stack does; this lifts the scheduler hunk out of it against `dev` so the E2E flakes it causes stop before tomorrow's `dev` → `master` release merge.

`Scheduler.run_service()` ran the search-embedding backfill synchronously before calling `AppService.run_service()`, and that call is what starts the event loop `uvicorn.serve()` was already scheduled onto (`backend/util/service.py:513-521` builds the server on its own thread and schedules the coroutine onto `shared_event_loop`; the loop only spins inside `AppService.run_service()`). Port 8003 therefore stayed closed for the whole backfill. On a fresh stack the scheduler reports `Total: 5584 items without embeddings (0.0% coverage) - processing all` and takes about four and a half minutes, which is longer than an entire Playwright E2E run — so `GET /api/graphs/{id}/schedules` and `GET /api/schedules/followups` 500 with `All connection attempts failed` for the whole suite, and the Library agent page renders an error card in place of its run sidebar. In CI run 34233328612 the scheduler logged `Startup embedding backfill complete` at 13:52:18,905 and `Uvicorn running on http://0.0.0.0:8003` 76 ms later, having started the backfill at 13:47:41,810.

The existing six-hourly `ensure_embeddings_coverage` job now carries `next_run_time=datetime.now(timezone.utc)`, so the backfill still runs at startup — just on APScheduler's thread rather than ahead of the port binding. `misfire_grace_time=None` keeps a late run from being skipped as a misfire, and `coalesce=True` collapses a backlog into one run.

### Changes 🏗️

- `ensure_embeddings_coverage` job registration gains `next_run_time` (now), `misfire_grace_time=None` and `coalesce=True`.
- The synchronous `if self.register_system_tasks:` backfill block before `super().run_service()` is deleted.
- Two tests on `run_service`'s startup path: one asserts the backfill has not run by the time `AppService.run_service()` is reached, one asserts the job is registered due-now and un-skippable.

**Taken from #14350, and left out:** the hunk there reads `hours=0 if config.scheduler_startup_embedding_backfill else 6`, gating the immediate run on a setting that only exists on the unmerged stack. This PR is the ordering fix alone, so the setting is left out and the job is unconditionally due now — matching `dev`'s current behaviour, where the startup backfill is likewise unconditional. #14350 also touches `single-container/Dockerfile` and its entrypoint test; none of that is here.

**Known follow-up, not fixed here:** `.github/workflows/platform-fullstack-ci.yml:168` writes `SCHEDULER_STARTUP_EMBEDDING_BACKFILL=false` into `backend/.env`, and nothing on `dev` reads that name — `git grep` finds it in the workflow file and nowhere else, because the setting that would honour it lives only on the stack. CI has believed the startup backfill was off since that line was added on 2026-04-14 in #12682.

### Agents and large language models used

Claude Code with Claude Opus 5.

### Verified

I ran both new tests against `dev`'s ordering and against this branch: red then green, `assert 1 == 0` on the readiness ordering and `KeyError: 'next_run_time'` on the registration. I also ran the whole `scheduler_unit_test.py` (95 passed), `backend/util/architecture_test.py`, `blocks/test/test_block.py` (1647 passed, 84 skipped) and the full `backend/executor` suite (512 passed, 6 failed) — the six are pre-existing `manager_test.py` failures on this machine (402 insufficient balance), which I confirmed by restoring both changed files to their `dev` contents and watching the same six fail. Separately I reproduced the port-binding delay against the real `AppService` class: 6.27 s to accept with the work on the pre-`run_service()` path, 0.25 s with it moved off, while the work still runs. Logs in a comment below.

I have not run the Playwright E2E suite locally; the CI E2E job on this PR is the check for that.

#### For code changes:
- [x] I have clearly listed my changes in the PR description
- [x] I have made a test plan
- [x] I have tested my changes according to the test plan:
  - [x] Both new tests fail on `dev`'s ordering and pass on this branch
  - [x] `scheduler_unit_test.py`, `architecture_test.py`, `blocks/test/test_block.py`, `backend/executor` — no new failures
  - [x] Port-binding delay reproduced and removed against the production `AppService`
  - [ ] Full-stack CI E2E on this PR

### CODE REVIEW

File: autogpt_platform/backend/backend/executor/scheduler.py [hunk 1]
```diff
@@ -1889,8 +1889,13 @@ def run_service(self):
                 id="ensure_embeddings_coverage",
                 trigger="interval",
                 hours=6,
+                # Due now rather than called inline below: run_service() is what
+                # starts the event loop uvicorn binds the RPC port on.
```
[the remark this answers is not in the document]
Comment (Pwuts) [reply autogpt_platform/backend/backend/executor/scheduler.py:1893]: 🤖 Not changing this: the ordering that caused the outage is same-thread, and this dispatch is on another thread.

`BackgroundScheduler.start()` returns immediately, so the main thread reaches `super().run_service()` microseconds later however soon APScheduler dispatches the job. The backfill then runs on an APScheduler worker doing blocking RPC I/O against database_manager (`get_database_manager_client()`), touching neither `shared_event_loop` nor the main thread. What broke readiness on `dev` is that the main thread itself sat inside `ensure_embeddings_coverage()` for ~4.5 minutes and so never reached the call that starts the loop `uvicorn.serve()` was scheduled onto.

Measured in exactly the case you describe. The repro in the evidence comment subclasses the real `AppService`, and its `fixed` mode fires the six seconds of work on a `threading.Timer(0.0, …)` *before* calling `super().run_service()` — the work is already running when readiness begins. The port accepts at 0.25 s:

```
[ 0.03s] startup work: begin
INFO  Uvicorn running on http://localhost:18099
[ 0.25s] PORT 18099 ACCEPTING
```

(the process exits there, with the work still running). Gating the first backfill on RPC readiness would re-couple the two for no gain.

On the test: it asserts that `run_service()` does not call the backfill inline, which is the regression — it fails on `dev` with `assert 1 == 0`, and `test_embedding_backfill_is_registered_to_run_immediately` fails there with `KeyError: 'next_run_time'`. Asserting *when* a real `BackgroundScheduler` dispatches a due job would be testing APScheduler's own threading contract and would make a unit test timing-dependent.

**Review by ntindle — APPROVED** [review]
